ledger/ledger-on-memory/src/app/scala/com/daml/ledger/on/memory/Main.scala

```diff
@@ -29,17 +29,26 @@ import com.daml.resources.{ProgramResource, ResourceOwner}
 import scopt.OptionParser
 
 object Main {
+  private val runnerName = "In-Memory Ledger"
   def main(args: Array[String]): Unit = {
     val resource = for {
       dispatcher <- dispatcherOwner
       sharedState = InMemoryState.empty
       factory = new InMemoryLedgerFactory(dispatcher, sharedState)
-      runner <- new Runner("In-Memory Ledger", factory).owner(args)
+      runner <- new Runner(runnerName, factory).owner(args)
     } yield runner
-
     new ProgramResource(resource).run()
   }
 
+  // Utily if you want to spin this up as a library.
+  def owner(config: Config[ExtraConfig]): ResourceOwner[Unit] =
```

Comment:
Can you use this in the `main` function above?

Reply:
Not easily, see PR description. You need the factory to parse the config so you end up having to split it up and at that point I think it just gets more convoluted.